content/en/blog/falco-driverkit-debian-docker.md

```diff
@@ -0,0 +1,516 @@
+2022-07-26T13:00:0+0000: Falco version 0.32.1
+2022-07-26T13:00:00+0000: Falco initialized with configuration file /etc/falco/falco.yaml
+2022-07-26T13:00:00+0000: Loading rules from file /etc/falco/falco_rules.yaml:
+2022-07-26T13:00:00+0000: Loading rules from file /etc/falco/falco_rules.local.yaml:
+2022-07-26T13:00:00+0000: Starting internal webserver, listening on port 8765
+2022-07-26T13:00:00+0000: Unable to load the driver.
+2022-07-26T13:00:00+0000: Runtime error: error opening device /host/dev/falco0. Make sure you have root credentials and that the falco module is loaded.. Exiting.
+```
+
+There are some important reads from this output:
+* The driver version this image tries to load is the **`2.0.0+driver`**. This information will be really useful when we need to compile and share the driver with the *falco container*.
+* By default, the container will look for a kernel module. It is possible to switch to an *eBPF probe* by using an environment variable, as you'll see later in this post.
+* The `falco-driver-loader` script always removes the driver from memory and tries to load a current one. This is done for security reasons and the way to avoid that is not running this script when creating the container. More on this later, too.
+* After looking in the system for a previously installed driver, the script tries to download it from the URL `https://download.falco.org`. Unfortunately, it doesn't seem to be able to find it and falls back to the local compilation method.
+* When the script tries to compile the driver inside the container, it doesn't succeed because we haven't fulfilled one important prerequisite: installing the kernel headers on the host machine. In this post, we won't address that method but you can always refer to the documentation.
+
+### Using Falco Driverkit
+
+As mentioned, there are different ways to obtain a valid kernel: downloading it from `https://download.falco.org`, compiling it via the `falco-driver-loader` script, or the method we'll explain here: using `driverkit`.
+
+We don't intend this post to be an exhaustive guide to `driverkit`. That's also why we've chosen a relatively easy and tested target operating system: Debian.
+
+First of all, we need the `driverkit` tool which we'll compile ourselves. We can download the source code from `https://github.com/falcosecurity/driverkit`.
+
+When compiling a tool, we like using a temporary container. In this case, we'll start our container using the `docker.io/golang:1.17` image and a `sleep` process until we're done. The `./driverkit` directory will help us to extract the binary to the host filesystem. Feel free to use any other method you prefer, like `docker cp`.
+
+```bash
+# This directory will contain the driverkit binary
+# once it is compiled inside the container
+
+$ mkdir ./driverkit
+
+# Container with Go tools
+$ docker run --rm -d --name golang-compiler \
```

Comment:
`--rm` is not compatibe with `-d`

Reply:
Could it be an old behavior? I tested it again and it worked.